recipes/openvino/all/conanfile.py

```diff
@@ -0,0 +1,381 @@
+
+    @property
+    def _target_arm(self):
+        return "arm" in self.settings.arch
+
+    @property
+    def _target_x86_64(self):
+        return self.settings.arch == "x86_64"
+
+    @property
+    def _gna_option_available(self):
+        return self.settings.os in ["Linux", "Windows"] and self._target_x86_64 and Version(self.version) < "2024.0.0"
+
+    @property
+    def _gpu_option_available(self):
+        return self.settings.os != "Macos" and self._target_x86_64
+
+    @property
+    def _preprocessing_available(self):
+        return Version(self.version) < "2023.2.0"
+
+    @property
+    def _onnx_version(self):
+        if Version(self.version) == "2023.1.0":
+            return "1.13.1"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "clang": "9",
+            "apple-clang": "11",
+            "Visual Studio": "16",
+            "msvc": "192",
+        }
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version]["openvino"], strip_root=True)
+        get(self, **self.conan_data["sources"][self.version]["onednn_cpu"], strip_root=True,
+            destination=f"{self.source_folder}/src/plugins/intel_cpu/thirdparty/onednn")
+        get(self, **self.conan_data["sources"][self.version]["mlas"], strip_root=True,
+            destination=f"{self.source_folder}/src/plugins/intel_cpu/thirdparty/mlas")
+        get(self, **self.conan_data["sources"][self.version]["arm_compute"], strip_root=True,
+            destination=f"{self.source_folder}/src/plugins/intel_cpu/thirdparty/ComputeLibrary")
+        get(self, **self.conan_data["sources"][self.version]["onednn_gpu"], strip_root=True,
+            destination=f"{self.source_folder}/src/plugins/intel_gpu/thirdparty/onednn_gpu")
+        apply_conandata_patches(self)
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if not self._gpu_option_available:
+            del self.options.enable_gpu
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+            if self._protobuf_required:
+                # we need to use static protobuf to overcome potential issues with multiple registrations inside
+                # protobuf when frontends (implemented as plugins) are loaded multiple times in runtime
+                self.options["protobuf"].shared = False
+        # disable GPU plugin when clang is used; plugin has issues with static variables initialization
+        if self.settings.compiler == "clang" and self.options.get_safe("enable_gpu"):
+            self.options.enable_gpu = False
+
+    def build_requirements(self):
+        if self._target_arm:
+            self.tool_requires("scons/[>=4.2.0]")
```

Comment:
@RubenRBS @uilianries I though that version range was only allowed for cmake, openssl, libcurl and zlib.

Reply:
Correct, it was accepted by mistake!

Reply:
You're totally right. We're thinking about allowing it for every tool_require in the future, but we're not there yet, thanks for noticing - This PR got sidetracked and we merged before we should have, and things like this slipped by - Pinging @ilya-lavrenov for when the update to use arm_compute is done, could you also please fix this? Thanks!

Reply:
I supposed that for tools we can use ranges (and about it here https://github.com/conan-io/conan-center-index/pull/19681#discussion_r1326432158)

Ok, I will pin tools version as well in my nearest update (I plan to backport some patches from upstream to make recipe more future proof).

Reply:
fixed in https://github.com/conan-io/conan-center-index/pull/20245